test/jdk/javax/swing/JComponent/7154030/bug7154030.java

```diff
@@ -81,25 +84,36 @@ public void run() {
                     button.setOpaque(true);
                     button.setVisible(false);
                     desktop.add(button);
+                    desktop.setMinimumSize(new Dimension(300, 300));
+                    desktop.setMaximumSize(new Dimension(300, 300));
 
                     frame.setContentPane(desktop);
-                    frame.setSize(300, 300);
+                    frame.setMinimumSize(new Dimension(350, 350));
+                    frame.setMaximumSize(new Dimension(350, 350));
+                    frame.pack();
                     frame.setLocationRelativeTo(null);
                     frame.setVisible(true);
                     frame.setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
                 }
             });
 
-            robot.delay(1000);
             robot.waitForIdle(1000);
+            robot.delay(1000);
```

Comment:
I doubt that this is a test issue, 2 seconds was not enough to properly show the frame?

Reply:
The problem is that i tried to reproduce this behavior locally on Ubuntu running this test (and the simplified version of it) literally millions of times. The only time it fails is when it is run on mach5 and even then it fails very rarely - once per hundreds of runs. The changes here - especially forcing the resize - seems to eliminate the problem with mach5.

Reply:
Then it is a product bug, If the problem is unrelated to the performance and the root cause is in the wrong/broken final layout. Do we want to suggest the same thing if the user will get the same issue in their app?

Reply:
It might be a product bug. Or it might be something specific to the machines with mach5 system. Or to some specific machine having hardware issue (i noticed that in my case all failures happened on one specific Linux test system). That's the problem. I have a simplified test case that should check for that specific problem and i'm running a dedicated Linux host in my home office with the same OS version than problematic machine that does nothing but runs this test in interactive mode for days. Today only i ran this test over 700000 times - 0 failures. If i would have a way to at least semi-reliable reproduce the problem i would create a new bug and would debug and fix it. But i don't. And the changes i made is simply a way to let this test to run even on the problematic host because the problem that happens have nothing to do with the functionality related to the test itself.